packages/engine/src/interaction/systems/GrabbableSystem.tsx

```diff
@@ -372,11 +373,22 @@ const execute = () => {
    */
   if (isClient)
     for (const entity of grabbableQuery.enter()) {
-      addInteractableUI(entity, createInteractUI(entity, grabbableInteractMessage), onGrabbableInteractUpdate)
+      setCallback(entity, 'grabCallback', () => grabCallback(entity))
+      setComponent(entity, InteractableComponent, {
+        label: grabbableInteractMessage,
+        callbacks: [
+          {
+            callbackID: 'grabCallback',
+            target: null
+          }
+        ]
+      })
+
+      //addInteractableUI(entity, createInteractUI(entity, grabbableInteractMessage), onGrabbableInteractUpdate)
```

Comment:
onGrabbableInteractUpdate callback is never used if this stays commented out. Is it beneficial to pass these update functions as callbacks? It seems to result in some duplicate code between the grabbable update function and the interactable update function.

Reply:
taken care of